src/appleseed/renderer/kernel/rendering/final/adaptivepixelrenderer.cpp

```diff
@@ -86,10 +83,11 @@ namespace
             ISampleRendererFactory*     factory,
             const ParamArray&           params,
             const size_t                thread_index)
-          : m_params(params)
+          : PixelRendererBase(frame, thread_index, params)
+          , m_params(params)
           , m_sample_renderer(factory->create(thread_index))
         {
-            if (m_params.m_diagnostics)
+            if (m_base_params.m_diagnostics)
```

Comment:
It's quite problematic to have parameters in two places, `m_base_params` and `m_params`. I would suggest simply keeping `m_base_params` in the base class, and renaming it to `m_params`.